core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -697,12 +711,22 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
                 this.cancel();
             } catch (InterruptedException | RetriableException ex) {
                 throw ex;
+            } catch (CorruptIndexException ex) {
+                logger.error("Error occurred while copying log segments. Index appeared to be corrupted for partition: {} ", topicIdPartition, ex);
+                segmentCopyFailures++;
```

Comment:
We can re-create index from the segment in case it is corrupted. This would prevent need of any operator intervention. It will block uploading to TS for a time duration during which index is being reconstructed but the trade-off in favour of self-recovery is worth it in my opinion.